marlowe-playground-client/src/Types.purs

```diff
@@ -254,6 +255,7 @@ type MarloweState
     , editorWarnings :: Array IMarker
     , holes :: Holes
     , payments :: Array Payment
+    , currentTransactionInput :: Maybe TransactionInput
```

Comment:
Why do we need `currentTransactionInput`? I understand it is essentially: `stateToTxInput`. We may want to get rid of `state` and `pendingInputs` instead. If we have duplicated information then we run the risk it will go out of sync and we will see glitches in the future.

Reply:
`currentTransactionInput` is historical, `state` is current and `pendingInputs` is future. Perhaps I should rename `currentTransactionInput` to `lastTransactionInput`? I think it's possible to get rid of `pendingInputs` but not `state` or `currentTransactionInput`.

Reply:
Sorry I meant `slot` not `state`

Reply:
another argument for the current method is that the rendering code should be static, any calculations and data manipulations should be done against the current state in the pre-render phase. It is, of course, a bit subjective as to whether `pendingInputs` counts or whether it is just a rendering of the state.

Reply:
same argument goes for `slot` although it seems a stronger argument for slot as you could easily use a lens for that. I think I'm going to leave this as is but if _you_ want to change it in any other PR then I would be happy to approve it.

Reply:
So I guess the question is. Is `currentTransactionInput` ever accessed at a time where its value cannot be inferred from `pendingInputs` and `slot`. If not I think we should get rid of one of them to keep things simple

Reply:
the idea is to try to keep the rendering code as clean and function free as possible. Not that it's particularly good at the moment but we don't want to make it worse.